update systemd-homed policy

zpytela opened the pull request:
Together with that, label /var/cache/systemd with systemd_cache_t.

Resolves: rhbz#2036108

zpytela committed:
Allow aide connect to systemd-homed over a unix socket

The commit addresses the following AVC denial:

type=AVC msg=audit(1727684243.591:2741): avc:  denied  { connectto } for  pid=6612 comm="aide" path="/run/systemd/userdb/io.systemd.Home" scontext=system_u:system_r:aide_t:s0-s0:c0.c1023 tcontext=system_u:system_r:systemd_homed_t:s0 tclass=unix_stream_socket permissive=0

Resolves: rhbz#2315635

zpytela committed:
Allow abrt-dump-journal-core connect to systemd-homed over a unix socket

The commit addresses the following AVC denial:
type=AVC msg=audit(1727748741.793:302): avc:  denied  { connectto } for  pid=2471 comm="abrt-dump-journ" path="/run/systemd/userdb/io.systemd.Home" scontext=system_u:system_r:abrt_dump_oops_t:s0 tcontext=system_u:system_r:systemd_homed_t:s0 tclass=unix_stream_socket permissive=1
type=SYSCALL msg=audit(1727748741.793:302): arch=x86_64 syscall=connect success=yes exit=0 a0=4b a1=7ffdcd4323b0 a2=26 a3=0 items=1 ppid=1 pid=2471 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm=abrt-dump-journ exe=/usr/bin/abrt-dump-journal-core subj=system_u:system_r:abrt_dump_oops_t:s0 key=(null)
type=PATH msg=audit(1727748741.793:302): item=0 name=/run/systemd/userdb/io.systemd.Home inode=3258 dev=00:1b mode=0140666 ouid=0 ogid=0 rdev=00:00 obj=system_u:object_r:systemd_userdbd_runtime_t:s0 nametype=NORMAL cap_fp=0 cap_fi=0 cap_fe=0 cap_fver=0 cap_frootid=0

Resolves: rhbz#2315813

zpytela commented:
@richiedaze FYI - it seems people started to use the feature.

zpytela committed:
Allow systemd-homed get attributes of a tmpfs filesystem

The commit addresses the following AVC denial:
type=AVC msg=audit(1727530369.503:330): avc:  denied  { getattr } for  pid=3418 comm="systemd-homed" name="/" dev="tmpfs" ino=1 scontext=system_u:system_r:systemd_homed_t:s0 tcontext=system_u:object_r:tmpfs_t:s0 tclass=filesystem permissive=0

Resolves: rhbz#2315462

zpytela committed:
Label /var/cache/systemd/home with systemd_homed_cache_t

Label /var/cache/systemd/home with systemd_homed_cache_t and allow
systemd-homed and systemd-homework read files in there.
Together with that, label /var/cache/systemd with systemd_cache_t.

Resolves: rhbz#2036108

richiedaze commented:
@zpytela 

Upgrading rawhide, updates systemd to version 256, which introduced new features to systemd-homed. 

I have adjusted in these new features on my testing branch, but I am still testing these features before submitting a pull request.

zpytela committed:
Allow boothd connect to systemd-homed over a unix socket

The commit addresses the following AVC denial:
type=PROCTITLE msg=audit(10/03/2024 18:41:15.337:2703) : proctitle=/usr/sbin/boothd daemon -S -c /etc/booth/booth.conf
type=PATH msg=audit(10/03/2024 18:41:15.337:2703) : item=0 name=/run/systemd/userdb/io.systemd.Home inode=1758 dev=00:1b mode=socket,666 ouid=root ogid=root rdev=00:00 obj=system_u:object_r:systemd_userdbd_runtime_t:s0 nametype=NORMAL cap_fp=none cap_fi=none cap_fe=0 cap_fver=0 cap_frootid=0
type=SOCKADDR msg=audit(10/03/2024 18:41:15.337:2703) : saddr={ saddr_fam=local path=/run/systemd/userdb/io.systemd.Home }
type=SYSCALL msg=audit(10/03/2024 18:41:15.337:2703) : arch=x86_64 syscall=connect success=no exit=EACCES(Permission denied) a0=0x9 a1=0x7ffc5f88f670 a2=0x26 a3=0x0 items=1 ppid=1 pid=63022 auid=unset uid=root gid=root euid=root suid=root fsuid=root egid=root sgid=root fsgid=root tty=(none) ses=unset comm=boothd exe=/usr/sbin/boothd subj=system_u:system_r:boothd_t:s0 key=(null)
type=AVC msg=audit(10/03/2024 18:41:15.337:2703) : avc:  denied  { connectto } for  pid=63022 comm=boothd path=/run/systemd/userdb/io.systemd.Home scontext=system_u:system_r:boothd_t:s0 tcontext=system_u:system_r:systemd_homed_t:s0 tclass=unix_stream_socket permissive=0

zpytela committed:
Allow login_userdomain connect to systemd-homed over a unix socket

The commit addresses the following AVC denial:
type=PROCTITLE msg=audit(10/04/2024 08:43:54.462:5944) : proctitle=id
type=PATH msg=audit(10/04/2024 08:43:54.462:5944) : item=0 name=/run/systemd/userdb/io.systemd.Home inode=1752 dev=00:1b mode=socket,666 ouid=root ogid=root rdev=00:00 obj=system_u:object_r:systemd_userdbd_runtime_t:s0 nametype=NORMAL cap_fp=none cap_fi=none cap_fe=0 cap_fver=0 cap_frootid=0
type=SOCKADDR msg=audit(10/04/2024 08:43:54.462:5944) : saddr={ saddr_fam=local path=/run/systemd/userdb/io.systemd.Home }
type=SYSCALL msg=audit(10/04/2024 08:43:54.462:5944) : arch=x86_64 syscall=connect success=no exit=EACCES(Permission denied) a0=0x9 a1=0x7ffe279c6df0 a2=0x26 a3=0x563dd3733740 items=1 ppid=161562 pid=161563 auid=user11355 uid=user11355 gid=user11355 euid=user11355 suid=user11355 fsuid=user11355 egid=user11355 sgid=user11355 fsgid=user11355 tty=pts2 ses=93 comm=id exe=/usr/bin/id subj=guest_u:guest_r:guest_t:s0 key=(null)
type=AVC msg=audit(10/04/2024 08:43:54.462:5944) : avc:  denied  { connectto } for  pid=161563 comm=id path=/run/systemd/userdb/io.systemd.Home scontext=guest_u:guest_r:guest_t:s0 tcontext=system_u:system_r:systemd_homed_t:s0 tclass=unix_stream_socket permissive=0
